### PR TITLE
Add C API functions to build list/map types and read map types

### PR DESCRIPTION
**Pull request by metab0t**
Add several C API functions to build list/map types and read map types.

If the naming and implementation are approved, I will add docstrings to these API functions in `duckdb.h`.

### CODE REVIEW

File: src/include/duckdb.h [hunk 1]
```diff
@@ -1000,6 +1000,10 @@ This should not be used with `DUCKDB_TYPE_DECIMAL`.
 */
 DUCKDB_API duckdb_logical_type duckdb_create_logical_type(duckdb_type type);
 
+DUCKDB_API duckdb_logical_type duckdb_create_list_logical_type(duckdb_logical_type type);
+DUCKDB_API duckdb_logical_type duckdb_create_map_logical_type(duckdb_logical_type key_type,
```
Comment (Mytherin) [inline src/include/duckdb.h:1004]: `duckdb_create_map_type`

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Looks good. Just one suggestion to rename the `create` functions to be consistent with the decimal create function, and add documentation as you mentioned. Note that for the comments it is important the exact format is followed as this is used to auto-generate documentation for the website.

File: test/api/capi/test_capi_data_chunk.cpp [hunk 2]
```diff
@@ -10,6 +10,31 @@ TEST_CASE("Test Logical Types C API", "[capi]") {
 	duckdb_destroy_logical_type(&type);
 	duckdb_destroy_logical_type(&type);
 
+	// list type
+	duckdb_logical_type elem_type = duckdb_create_logical_type(DUCKDB_TYPE_INTEGER);
+	duckdb_logical_type list_type = duckdb_create_list_type(elem_type);
+	REQUIRE(duckdb_get_type_id(list_type) == DUCKDB_TYPE_LIST);
+	duckdb_logical_type elem_type_dup = duckdb_list_type_child_type(list_type);
+	REQUIRE(duckdb_get_type_id(elem_type_dup) == duckdb_get_type_id(elem_type));
```
Comment (hawkfish) [inline test/api/capi/test_capi_data_chunk.cpp:18]: Maybe assert that `elem_type != elem_type_dup`?

**Review by hawkfish — APPROVED** [review]

**Review by Mytherin — APPROVED** [review]
Thanks for the updates! Looks good to me.

**Pull request merged**